app/client/components/identity/idapi/fetch.ts

```diff
@@ -0,0 +1,41 @@
+export const APIFetch = (baseUrl: string) => async (
+  url: string,
+  options?: RequestInit
+): Promise<any> => {
+  const response = await fetch(baseUrl + url, options);
+  if (!response.ok) {
+    let err;
```

Comment:
Could this be moved into a seperate function to make this whole block more readable ?